### PR TITLE
[Sparse] Elementwise ops for DiagMatrix and Elementwise Power for SparseMatrix

### PR DESCRIPTION
**Pull request by mufeili**
## Checklist
Please feel free to remove inapplicable items for your PR.
- [x] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [x] I've leverage the [tools](https://docs.google.com/document/d/1iHyj7zlmygKSk5gBPsqIqL5ASPzJSPREaNT_QdsiYA4/edit) to beautify the python and c++ code.
- [x] The PR is complete and small, read the [Google eng practice (CL equals to PR)](https://google.github.io/eng-practices/review/developer/small-cls.html) to understand more about small PR. In DGL, we consider PRs with less than 200 lines of core code change are small (example, test and documentation could be exempted).
- [x] All changes have test coverage
- [x] Code is well-documented
- [x] To the best of my knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change
- [x] Related issue is referred in this PR
- [ ] If the PR is for a new model/paper, I've updated the example index [here](../examples/README.md).

### CODE REVIEW

File: python/dgl/mock_sparse2/elementwise_op_diag.py [hunk 1]
```diff
@@ -146,41 +148,50 @@ def diag_div(D1: DiagMatrix, D2: Union[DiagMatrix, float]) -> DiagMatrix:
...
     shape=(3, 3))
     """
-    if isinstance(D1, DiagMatrix) and isinstance(D2, DiagMatrix):
-        assert (
-            D1.shape == D2.shape
-        ), "The shape of diagonal matrix D1 {} and" "D2 {} must match".format(
-            D1.shape, D2.shape
+    if isinstance(D2, DiagMatrix):
+        assert D1.shape == D2.shape, (
+            "The shape of diagonal matrix D1 "
```
Comment (frozenbugs) [inline python/dgl/mock_sparse2/elementwise_op_diag.py:153]: nit: let's make full use of the line until 80 chars before change line.

File: python/dgl/mock_sparse2/elementwise_op_diag.py [hunk 2]
```diff
@@ -146,41 +148,50 @@ def diag_div(D1: DiagMatrix, D2: Union[DiagMatrix, float]) -> DiagMatrix:
...
+            f"{D1.shape} and D2 {D2.shape} must match."
         )
         return diag(D1.val / D2.val, D1.shape)
-    return diag(D1.val / D2, D1.shape)
+    elif isinstance(D2, (float, int)):
+        assert D2 != 0, "Division by zero is not allowed."
+        return diag(D1.val / D2, D1.shape)
 
+    raise RuntimeError(
+        "Elementwise division between "
```
Comment (frozenbugs) [inline python/dgl/mock_sparse2/elementwise_op_diag.py:162]: ditto.

File: python/dgl/mock_sparse2/elementwise_op_diag.py [hunk 3]
```diff
@@ -146,41 +148,50 @@ def diag_div(D1: DiagMatrix, D2: Union[DiagMatrix, float]) -> DiagMatrix:
...
-    D2 : DiagMatrix
+    D1 : DiagMatrix
         Diagonal matrix
+    D2 : float or int
+        Scalar value
     """
     raise RuntimeError(
-        "Elementwise subtraction between {} and {} is not "
-        "supported.".format(type(D1), type(D2))
+        "Elementwise division of "
```
Comment (frozenbugs) [inline python/dgl/mock_sparse2/elementwise_op_diag.py:179]: ditto.

File: python/dgl/mock_sparse2/elementwise_op_diag.py [hunk 4]
```diff
@@ -120,15 +120,17 @@ def diag_mul(
...
 
     Parameters
     ----------
     D1 : DiagMatrix
         Diagonal matrix
-    D2 : DiagMatrix or scalar
-        Diagonal matrix or scalar value
+    D2 : DiagMatrix or float or int
+        Diagonal matrix or scalar value. If :attr:`D2` is a DiagMatrix,
+        division is only applied to the diagonal elements.
```
Comment (frozenbugs) [inline python/dgl/mock_sparse2/elementwise_op_diag.py:133]: ditto.

**Review by frozenbugs — APPROVED** [review]

**Comment by frozenbugs** [comment]
@dgl-bot

**Review by czkkkkkk — APPROVED** [review]

**Pull request merged**